packages/core/useMediaQuery/index.ts

```diff
@@ -21,39 +21,44 @@ export function useMediaQuery(query: MaybeRefOrGetter<string>, options: Configur
   let mediaQuery: MediaQueryList | undefined
   const matches = ref(false)
 
+  const handler = (event: MediaQueryListEvent) => {
+    matches.value = event.matches
+  }
+
   const cleanup = () => {
     if (!mediaQuery)
       return
     if ('removeEventListener' in mediaQuery)
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      mediaQuery.removeEventListener('change', update)
+      mediaQuery.removeEventListener('change', handler)
     else
       // @ts-expect-error deprecated API
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      mediaQuery.removeListener(update)
+      mediaQuery.removeListener(handler)
   }
 
-  const update = () => {
+  const stopWatch = watchEffect(() => {
     if (!isSupported.value)
       return
 
     cleanup()
 
-    mediaQuery = window!.matchMedia(toRef(query).value)
+    mediaQuery = window!.matchMedia(toValue(query))
     matches.value = !!mediaQuery?.matches
 
     if (!mediaQuery)
       return
```

Comment:
It should be possible to assert that if `isSupported.value = true`, then `window.matchMedia` always returns a `MediaQueryList` unless there exists a noncompliant implementation that this aims to handle.